pkg/apis/provisioning/v1alpha1/provisioner.go

```diff
@@ -40,9 +63,28 @@ type ProvisionerSpec struct {
 	// InstanceTypes constraints which instances types will be used for nodes
 	// launched by the Provisioner. If unspecified, supports all types. Cannot
 	// be specified if label "node.kubernetes.io/instance-type" is specified.
+	// +optional
 	InstanceTypes []string `json:"instanceTypes,omitempty"`
+	// Architecture constraints the underlying node architecture
```

Comment:
typo `constraints` -> `constrains`? Also might be worth stating (although fairly obvious) that if not specified any architecture (and maybe reassure people that if they want a single Provisioner to handle both aarch64/amd64 (say), that would actually work?)